### PR TITLE
Get gpx file import working for ut version

### PR DESCRIPTION
**Pull request by StefWe**
#324 Implements the GPX file selector based on UT's Content Hub. Has been successfully tested by various persons from the community.

### CODE REVIEW

File: qml/platform.ubports/FileSelectorPL.qml [hunk 1]
```diff
@@ -18,17 +18,69 @@
...
+import QtQuick 2.9
+import QtQuick.Controls 2.2
+import Ubuntu.Content 1.3
+import "."
 
-Item {
+Page {
     id: fs
-
+
```
Comment (rinigus) [inline qml/platform.ubports/FileSelectorPL.qml]: please remove whitespace

File: qml/platform.ubports/FileSelectorPL.qml [hunk 2]
```diff
@@ -18,17 +18,69 @@
...
     signal selected
+    signal cancel()    
+    
+	property var activeTransfer
+	property var url
+	property var handler
+	property var contentType
+
+    header: HeaderBarImpl {
+        title: i18n.tr("Choose")
```
Comment (rinigus) [inline qml/platform.ubports/FileSelectorPL.qml]: use `app.tr` instead

File: qml/platform.ubports/FileSelectorPL.qml [hunk 3]
```diff
@@ -18,17 +18,69 @@
...
+        onPeerSelected: {
+            peer.selectionType = ContentTransfer.Single
+            fs.activeTransfer = peer.request()
+            fs.activeTransfer.stateChanged.connect(function() {
+				if (fs.activeTransfer.state === ContentTransfer.InProgress) {
+					console.log("In progress");
+					fs.activeTransfer.items = fs.activeTransfer.items[0].url = url;
+					fs.activeTransfer.state = ContentTransfer.Charged;
+				}
+                if (fs.activeTransfer.state === ContentTransfer.Charged) {
```
Comment (rinigus) [inline qml/platform.ubports/FileSelectorPL.qml]: some alignment issues in this function

File: qml/platform.ubports/FileSelectorPL.qml [hunk 4]
```diff
@@ -18,17 +18,69 @@
...
+					console.log("Charged");
+                    selectedFilepath = fs.activeTransfer.items[0].url;
+                    selectedFilepath = selectedFilepath.replace("file://", "");
+                    fs.selected();
+                    app.pages.pop(Qt.resolvedUrl("../qml/platform/FileSelectorPL.qml"));
```
Comment (rinigus) [inline qml/platform.ubports/FileSelectorPL.qml]: why not `app.pages.pop()`?

File: qml/platform.ubports/FileSelectorPL.qml [hunk 5]
```diff
@@ -18,17 +18,69 @@
...
+                }
+            })
+        }
+
+        onCancelPressed: {
+            pageStack.pop()
```
Comment (rinigus) [inline qml/platform.ubports/FileSelectorPL.qml]: we don't call `pageStack` directly throughout the code. Let's keep it and use `app.pages.pop()` instead.

File: qml/platform.ubports/FileSelectorPL.qml [hunk 6]
```diff
@@ -18,17 +18,69 @@
...
+            peer.selectionType = ContentTransfer.Single
+            fs.activeTransfer = peer.request()
+            fs.activeTransfer.stateChanged.connect(function() {
+				if (fs.activeTransfer.state === ContentTransfer.InProgress) {
+					console.log("In progress");
+					fs.activeTransfer.items = fs.activeTransfer.items[0].url = url;
+					fs.activeTransfer.state = ContentTransfer.Charged;
+				}
+                if (fs.activeTransfer.state === ContentTransfer.Charged) {
+					console.log("Charged");
```
Comment (rinigus) [inline qml/platform.ubports/FileSelectorPL.qml]: I guess this can be dropped

File: qml/platform.ubports/FileSelectorPL.qml [hunk 7]
```diff
@@ -18,17 +18,69 @@
...
+        id: transferHint
+        anchors.fill: parent
+        activeTransfer: fs.activeTransfer
+    }
 
-    // todo: if open function is not needed, drop special consideration of
-    // of FileSelectorPL in platform's StackPL
+    Component {
+        id: resultComponent
+        ContentItem {}
```
Comment (rinigus) [inline qml/platform.ubports/FileSelectorPL.qml]: is this component used anywhere?

File: qml/platform.ubports/FileSelectorPL.qml (hunk 6 above)
Comment (StefWe) [reply qml/platform.ubports/FileSelectorPL.qml]: you mean the log output?

Comment (rinigus) [reply qml/platform.ubports/FileSelectorPL.qml]: yes

**Comment by rinigus** [comment]
Thank you very much!

**Comment by StefWe** [comment]
Thank you to be patient and all the help!

**Pull request merged**